src/core/ngx_memcache.c

```diff
@@ -1984,6 +2002,17 @@ static ngx_uint_t ngx_memcache_perl_hash (u_char *key, size_t len)
     return h;
 }
 
+static ngx_str_t ngx_sha256_hash (ngx_pool_t* p, u_char* key, size_t len)
+{
+    ngx_str_t hashed_key;
+    u_char * digest = ngx_pnalloc(p, SHA256_KEY_LENGTH);
+    SHA256(key, len, digest);
+    hashed_key.data = ngx_memcache_hexstr(digest, SHA256_KEY_LENGTH);
```

Comment:
If the allocation fails and returns null this or the next line will :boom:

So the call site's null check won't be reached. I see similar on the other md5 method in this line so probably not a big deal.

Reply:
Ok. understood your point ignoring boom for now as done in md5 !